src/EventService.Business/Commands/Event/CreateEventCommand.cs

```diff
@@ -0,0 +1,147 @@
+  private readonly IAccessValidator _accessValidator;
+  private readonly IResponseCreator _responseCreator;
+  private readonly IHttpContextAccessor _contextAccessor;
+  private readonly IEmailService _emailService;
+  private readonly IUserService _userService;
+
+  public CreateEventCommand(
+    IEventRepository repository,
+    IEventUserRepository eventUserRepository,
+    IEventCategoryRepository eventCategoryRepository,
+    ICreateEventRequestValidator validator,
+    IDbEventMapper eventMapper,
+    IDbEventUserMapper eventUserMapper,
+    IDbEventCategoryMapper eventCategoryMapper,
+    IAccessValidator accessValidator,
+    IResponseCreator responseCreator,
+    IHttpContextAccessor contextAccessor,
+    IUserService userService,
+    IEmailService emailService)
+  {
+    _eventRepository = repository;
+    _eventUserRepository = eventUserRepository;
+    _eventCategoryRepository = eventCategoryRepository;
+    _eventMapper = eventMapper;
+    _eventUserMapper = eventUserMapper;
+    _eventCategoryMapper = eventCategoryMapper;
+    _validator = validator;
+    _accessValidator = accessValidator;
+    _responseCreator = responseCreator;
+    _contextAccessor = contextAccessor;
+    _userService = userService;
+    _emailService = emailService;
+  }
+
+  private async Task SendInviteEmailsAsync(List<Guid> userIds, string eventName)
+  {
+    List<UserData> usersData = await _userService.GetUsersDataAsync(userIds);
+
+    if (usersData is null || !usersData.Any())
+    {
+      return;
+    }
+
+    foreach (UserData user in usersData)
+    {
+      await _emailService.SendAsync(
+        user.Email,
+        "Invite to event",
+        $"You have been invited to event {eventName}");
+    }
+  }
+
+  public async Task<OperationResultResponse<bool>> ExecuteAsync(CreateEventRequest request)
+  {
+    Guid senderId = _contextAccessor.HttpContext.GetUserId();
+    bool hasAddEditRemoveUsersRights = await _accessValidator.HasRightsAsync(senderId, Rights.AddEditRemoveUsers);
+    OperationResultResponse<bool> response = new();
+
+    if (!hasAddEditRemoveUsersRights)
+    {
+      return _responseCreator.CreateFailureResponse<bool>(HttpStatusCode.Forbidden);
+    }
+
+    if (request.Users.Distinct().Count() != request.Users.Count())
+    {
+      response.Errors = new List<string>() { "Some duplicate users have been removed from the list." };
+      request.Users = request.Users.Distinct().ToList();
+    }
+
+    request.CategoryIds = request.CategoryIds.Distinct().ToList();
+
+    ValidationResult validationResult = await _validator.ValidateAsync(request);
+    if (!validationResult.IsValid)
+    {
+      return _responseCreator.CreateFailureResponse<bool>(
+        HttpStatusCode.BadRequest,
+        validationResult.Errors.Select(er => er.ErrorMessage).ToList());
+    };
+
+    DbEvent dbEvent = _eventMapper.Map(request, senderId);
+
+    if ((dbEvent.Access == AccessType.Closed && !hasAddEditRemoveUsersRights)
```

Comment:
why you check it here, but not before validation?

Reply:
Deleted